arborist/server_test.go

```diff
@@ -1965,13 +1965,15 @@ func TestServer(t *testing.T) {
 		})
 
 		t.Run("RevokePolicy", func(t *testing.T) {
-			test := func(authzProvider string, expected bool, msg string) {
+			test := func(authzProvider string, httpStatusCode int, expected bool, msg string) {
 				w := httptest.NewRecorder()
 				url := fmt.Sprintf("/user/%s/policy/%s", username, policyName)
 				req := newRequest("DELETE", url, nil)
-				req.Header.Add("X-AuthZ-Provider", authzProvider)
+				if authzProvider != "" {
+					req.Header.Add("X-AuthZ-Provider", authzProvider)
+				}
 				handler.ServeHTTP(w, req)
-				if w.Code != http.StatusNoContent {
+				if w.Code != httpStatusCode {
 					httpError(t, w, "couldn't revoke policy")
```

Comment:
i think the error message should be updated to something like "revoke policy: expected status code x, got status code y" because "couldn't revoke policy" doesn't fit the new test cases